startup/90-functions.py

```diff
@@ -114,584 +248,854 @@ def print_det_keys(det_class):
-        'mQuestarX':mQuestarX.position,      
-        'mSigrayX':mSigrayX.position,    
-        'mSigrayY':mSigrayY.position,    
-        'mSigrayZ':mSigrayZ.position,    
-        'mSigrayPitch':mSigrayPitch.position,   
-        'mSigrayYaw':mSigrayYaw.position,     
-        'FastShutter':FastShutter.position, 
-        'RingCurrent':ring_current.get(),
-        'mHexapodsZ':mHexapodsZ.position,
-        'ePhi':ePhi.position,
-       }
 
-    ds.attrs = md
     
     
-    return ds
-
     
     
     
     
     
     
-"""  
-
-LEGACY:   
     
     
     
-def counter(det,exposure_time=1, take_dark=False, take_bright=True, num_dark = 3, num_bright = 2):
-
-    ds = xr.Dataset()
-
     
-    if take_dark:
-        set_detector(det,exposure_time=exposure_time,num_images=num_dark)
+    
+def gridscanner_1det(    
+                    det = None,
+                    det_acq_time = 2,
+                    include_dark = True,
+                    det_exposure_dark = 2,
+                    det_exposure_gridscan = 2,
+    
+                    include_flat=False,
+                    flat_motor=None,
+                    flat_motor_move=None,    
+
+                    #put slowest motor first
+                    motor1 = None,
+                    motor1_start = None,
+                    motor1_stop  = None,
+                    motor1_nstep = None,
+
+                    motor2 = None,
+                    motor2_start = None,
+                    motor2_stop  = None,
+                    motor2_nstep = None,
+
+                    snake=True,
+                    come_back = True,
+    
+                    ):
 
-#         laser_off()
-#         light1_off()
-#         light2_off()
 
+    if include_dark:
+        print('\n>>>> Taking dark\n') 
         beam_off()
-        uid_dark = RE(count([det],num=1))[0]
-        
-        if det.name == 'prosilica':
-            tiffs = get_tiff_list(hdr=db[-1])
-            t0 = fabio.open(tiffs[0]).data
-            img_dark = np.zeros((len(tiffs),t0.shape[0],t0.shape[1]))
-            for e,t in enumerate(tiffs):
-                img_dark[e,:,:] = fabio.open(tiffs[e]).data
-        else:
-            img_dark = np.array(list(db[-1].data('%s_image'%(det.name))))
-            
-
-        if len(img_dark.shape) == 4:
-            img_dark = img_dark.mean(axis=1)
-            img_dark = img_dark.mean(axis=0)
-        if len(img_dark.shape) == 3:
-            img_dark = img_dark.mean(axis=0)
-            
-        da_dark = xr.DataArray(data=img_dark.astype('float32'),
-                  coords=[np.arange(img_dark.shape[0]), np.arange(img_dark.shape[1])],
-                  dims=['pixel_y', 'pixel_x'],attrs=None
-                 )
-        ds['dark'] = da_dark
-        dark_taken = 'true'
-        
+        RE(configure_area_det(det,acq_time=det_acq_time,exposure=det_exposure_dark))
+        uid_dark = RE(count([det]),md={'task':'dark','acq_time':det_acq_time,'exposure':det_exposure_dark})[0]
```

Comment:
We do not recommend calling RE from inside functions. Better to make this function a plan with `yield from configure_area_det(...)`, etc. and pass it to `RE`.

Reply:
@mrakitin I will work on functions later. For now, removed those stuff